### PR TITLE
[NEXUS-8674] Eliminate SSL capabilities

### PR DESCRIPTION
**Pull request by adreghiciu**
https://issues.sonatype.org/browse/NEXUS-8674

### CODE REVIEW

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF342-1 :white_check_mark:

File: plugins/security/nexus-ssl-plugin/src/main/java/com/sonatype/nexus/ssl/plugin/internal/SMTPSessionParametersCustomizer.java [hunk 1]
```diff
@@ -53,15 +51,15 @@ public SMTPSessionParametersCustomizer(final TrustStore trustStore) {
 
   //@Override
   public Properties customize(final Properties params) {
-    final SSLContext sslContext = trustStore.getSSLContextFor(smtpTrustStoreKey());
     final String host = params.getProperty("mail.smtp.host");
     final String port = params.getProperty("mail.smtp.port");
-    if (sslContext != null && params.remove("mail.smtp.socketFactory.class") != null) {
+    if (Boolean.valueOf(params.getProperty("mail.smtp.ssl.useTrustStore"))
```
Comment (mcculls) [inline plugins/security/nexus-ssl-plugin/src/main/java/com/sonatype/nexus/ssl/plugin/internal/SMTPSessionParametersCustomizer.java]: I assume `mail.smtp.ssl.useTrustStore` is a nexus-specific property? (I don't see it listed in the table from https://javamail.java.net/nonav/docs/api/com/sun/mail/smtp/package-summary.html) If so then we should document this somewhere.

**Comment by mcculls** [comment]
+1

File: plugins/security/nexus-ssl-plugin/src/main/java/com/sonatype/nexus/ssl/plugin/internal/SMTPSessionParametersCustomizer.java (hunk 1 above)
Comment (adreghiciu) [inline plugins/security/nexus-ssl-plugin/src/main/java/com/sonatype/nexus/ssl/plugin/internal/SMTPSessionParametersCustomizer.java]: Is nexus specific. maybe we should use another name for this property.

**Comment by adreghiciu** [comment]
Manually merged https://github.com/sonatype/nexus-oss/commit/092fa413fc747fa5946b70dbbe7e1c0fbab0e424

**Pull request closed without merging**